packages/fast-development-site-react/src/components/site/index.tsx

```diff
@@ -295,6 +294,22 @@ class Site extends React.Component<ISiteProps & IManagedClasses<ISiteManagedClas
         locales: ["en", "en-rtl"]
     };
 
+    public static getDerivedStateFromProps(props: ISiteProps, state: ISiteState): Partial<ISiteState> | null {
+        if (props.activeTheme) {
+            return props.activeTheme !== state.theme ? { theme: props.activeTheme } : null;
```

Comment:
Rather than returning null.  It might be nice to consider having application logic to handle throwing exceptions rather than null statements.  The exception could be "BusinessExceptionNoTheme" for example.  Then it can be gracefully handled by the calling function and more reusable.  They might go more along the side of application instrumentation/exception management which I have in future planning.  But, just an idea.

Reply:
This is a React lifecycle hook that expects a null return value to signal no changes to state, so it's not really up to us as that is the API that React landed on.

Reply:
Thanks for the clarification.